### PR TITLE
AD-204581: Fixes for DSS CENHA - failure to retrieve subscriptions

### PR DESCRIPTION
**Pull request by hnandra-DFE**

### CODE REVIEW

File: NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs [hunk 1]
```diff
@@ -1,63 +1,45 @@
...
-
-            var query = client
-                ?.CreateDocumentQuery<Models.Subscriptions>(collectionUri)
-                .Where(x => x.CustomerId == customerId &&
-                            x.TouchPointId != senderTouchPointId &&
-                            x.Subscribe)
-                .AsDocumentQuery();
-
-            if (query == null)
+            if (customerId == null)
```
Comment (bmatthews-DFE) [inline NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs:20]: What about `sendTouchPointId`?

File: NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs [hunk 2]
```diff
@@ -1,63 +1,45 @@
...
+            {
                 return null;
+            }
+
+            var query = _container.GetItemLinqQueryable<Subscriptions>(true)
```
Comment (bmatthews-DFE) [inline NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs:25]: What does the `(true)` signify? Can you add a comment please :)

File: NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs [hunk 3]
```diff
@@ -1,63 +1,45 @@
...
-                var results = await query.ExecuteNextAsync<Models.Subscriptions>();
+                var results = await query.ReadNextAsync();
                 subscriptions.AddRange(results);
             }
 
             return subscriptions.Any() ? subscriptions : null;
         }
 
-        public async Task<ResourceResponse<Document>> CreateSubscriptionsAsync(Models.Subscriptions subscriptions)
+        public async Task<ItemResponse<Subscriptions>> CreateSubscriptionsAsync(Subscriptions subscriptions)
```
Comment (bmatthews-DFE) [inline NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs:40]: Can we remove this please - CENHA doesn't support creation of subscriptions - this (I believe to be) redundant

File: NCS.DSS.ContentEnhancer/Cosmos/Provider/IDocumentDBProvider.cs [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+using NCS.DSS.ContentEnhancer.Models;
 
 namespace NCS.DSS.ContentEnhancer.Cosmos.Provider
 {
     public interface IDocumentDBProvider
     {
-        Task<List<Models.Subscriptions>> GetSubscriptionsByCustomerIdAsync(Guid? customerId, string SenderTouchpointId);
-        Task<ResourceResponse<Document>> CreateSubscriptionsAsync(Models.Subscriptions subscriptions);
+        Task<List<Subscriptions>> GetSubscriptionsByCustomerIdAsync(Guid? customerId, string senderTouchpointId);
+        Task<ItemResponse<Subscriptions>> CreateSubscriptionsAsync(Subscriptions subscriptions);
```
Comment (bmatthews-DFE) [inline NCS.DSS.ContentEnhancer/Cosmos/Provider/IDocumentDBProvider.cs:9]: Ditto - remove line from this interface

File: NCS.DSS.ContentEnhancer.Tests/FunctionTests/QueueProcessorTests.cs
Comment (bmatthews-DFE) [inline NCS.DSS.ContentEnhancer.Tests/FunctionTests/QueueProcessorTests.cs:1]: We will need to test more edge cases to this, but it's a good start

**Review by bmatthews-DFE — CHANGES_REQUESTED** [review]
See comments

File: NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs (hunk 2 above)
Comment (hnandra-DFE) [reply NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs]: (true) is no longer required and has been removed.

**Comment by hnandra-DFE** [comment]
Comments resolved

File: NCS.DSS.ContentEnhancer.Tests/FunctionTests/QueueProcessorTests.cs
Comment (bmatthews-DFE) [inline NCS.DSS.ContentEnhancer.Tests/FunctionTests/QueueProcessorTests.cs:1]: All comments resolved, apart from this one - please can we add more unit tests?

**Review by bmatthews-DFE — CHANGES_REQUESTED** [review]
See comments

File: NCS.DSS.ContentEnhancer.Tests/FunctionTests/QueueProcessorTests.cs
Comment (bmatthews-DFE) [reply NCS.DSS.ContentEnhancer.Tests/FunctionTests/QueueProcessorTests.cs:1]: Unit tests added - resolving @hnandra-DFE

**Review by bmatthews-DFE — APPROVED** [review]
LGTM

AC #1:
image

AC #2:
image

**Review by bmatthews-DFE — APPROVED** [review]
Reapproval - see previous comments

**Review by ajones-DFE — COMMENTED** [review]
left comment on wrong repo
